library/alloc/src/collections/vec_deque/into_iter.rs

```diff
@@ -54,15 +55,16 @@ impl<T, A: Allocator> Iterator for IntoIter<T, A> {
     }
 
     #[inline]
-    fn advance_by(&mut self, n: usize) -> Result<(), usize> {
-        if self.inner.len < n {
+    fn advance_by(&mut self, n: usize) -> Result<(), NonZeroUsize> {
+        let rem = if self.inner.len < n {
             let len = self.inner.len;
             self.inner.clear();
-            Err(len)
+            len - n
```

Comment:
we're in a `len < n` block, so this looks wrong?
```suggestion
            n - len
```

(Maybe that means we need more tests for these too.)

Reply:
Good catch. I've added tests.